openmmtools/tests/test_sampling.py

```diff
@@ -2612,6 +2615,46 @@ def test_resume_velocities_from_legacy_storage(self):
                     state.velocities.value_in_unit_system(unit.md_unit_system) != 0
                 ), "At least some velocity in sampler state from new checkpoint is expected to different from zero."
 
+@pytest.fixture
+def download_nc_file(tmpdir):
+    FILE_URL = "https://github.com/user-attachments/files/17156868/ala-thr.zip"
+    MAX_RETRIES = 3
+    RETRY_DELAY = 2  # Delay between retries (in seconds)
+    file_name = os.path.join(tmpdir, "ala-thr.nc")
+    retries = 0
+    while retries < MAX_RETRIES:
+        try:
+          # Send GET request to download the file
+            response = requests.get(FILE_URL, timeout=20)  # Timeout to avoid hanging
+            response.raise_for_status()  # Raise HTTPError for bad responses (4xx/5xx)
+            with open(file_name, "wb") as f:
+                f.write(response.content)
+            # File downloaded successfully, break out of retry loop
+            break
+
+        except (requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
+            retries += 1
+            if retries >= MAX_RETRIES:
+                pytest.fail(f"Failed to download file after {MAX_RETRIES} retries: {e}")
+            else:
+                print(f"Retrying download... ({retries}/{MAX_RETRIES})")
+                time.sleep(RETRY_DELAY)  # Wait before retrying
+    yield file_name
+
+
+def test_pymbar_issue_419(download_nc_file):
+    from openmmtools.multistate import MultiStateReporter, MultiStateSamplerAnalyzer
+
+    n_iterations = 1000
+    reporter_file = download_nc_file
+    reporter = MultiStateReporter(reporter_file)
+    analyzer = MultiStateSamplerAnalyzer(reporter, max_n_iterations=n_iterations)
+    f_ij, df_ij = analyzer.get_free_energy()
+    # free energy
+    assert abs(f_ij[0, -1] - -52.00083148433459) < 1e-5
+    # error
+    assert abs(df_ij[0, -1] - 0.21365627649558516) < 1e-5
```

Comment:
The fun way to do this is pytest.approx if you've not tried it before!

Reply:
I've used `numpy.isclose` but never `pytest.approx` before

Reply:
It's the single value approximate, super fast and good error messages, have a go if you want!